### PR TITLE
[vsphere_host] Add error messages

### PR DESCRIPTION
**Pull request by koikonom**

### CODE REVIEW

File: vsphere/resource_vsphere_host.go [hunk 1]
```diff
@@ -657,7 +652,7 @@ func hostLockdownString(lockdownMode types.HostLockdownMode) (string, error) {
 		log.Printf("Found match for %s. Returning %s.", lockdownMode, modeString)
 		return modeString, nil
 	}
-	return "", fmt.Errorf("Unknwown Lockdown mode encountered")
+	return "", fmt.Errorf("unknwown Lockdown mode encountered")
```
Comment (aareet) [inline vsphere/resource_vsphere_host.go]: "Unknown"

File: vsphere/resource_vsphere_host.go [hunk 2]
```diff
@@ -642,7 +637,7 @@ func hostLockdownType(lockdownMode string) (types.HostLockdownMode, error) {
...
```
Comment (aareet) [inline vsphere/resource_vsphere_host.go]: "Unknown"

File: vsphere/resource_vsphere_host.go [hunk 3]
```diff
@@ -114,22 +114,20 @@ func resourceVsphereHostRead(d *schema.ResourceData, meta interface{}) error {
 			d.SetId("")
 			return nil
 		}
-		log.Printf("[DEBUG] Error while searching host %s. Error: %s", hostID, err)
-		return err
+		return fmt.Errorf("error while searching host %s. Error: %s ", hostID, err)
```
Comment (aareet) [inline vsphere/resource_vsphere_host.go]: Is it worth pulling the string "error while searching ...." into a local variable? Seems to be repeated a lot

**Comment by aareet** [comment]
@koikonom bump on the comments :)

File: vsphere/resource_vsphere_host.go (hunk 3 above)
Comment (koikonom) [reply vsphere/resource_vsphere_host.go]: Seems like an overkill, I'd rather come up with original error messages :)

**Comment by koikonom** [comment]
Thanks for the reminder! Take a look and if you're happy I can merge on Monday.

**Pull request merged**